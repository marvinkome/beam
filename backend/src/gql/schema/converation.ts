import { gql } from 'apollo-server-express'

export const conversationTypeDef = gql`
    type Conversation {
        id: ID!
        messages(first: Int, sort: Boolean, after: Int): [Message]
    }
`
