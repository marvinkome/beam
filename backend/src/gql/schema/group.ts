import { gql } from 'apollo-server-express'

export const groupTypeDefs = gql`
    type Role {
        name: String
    }

    type GroupUser {
        user: User
        role: Role
    }

    type Group {
        id: String
        name: String
        image: String
        interest: String
        location: String
        users: [GroupUser]

        isMember: Boolean
        lastMessage: Message
        messages(first: Int, after: Int): [Message]
    }
`
