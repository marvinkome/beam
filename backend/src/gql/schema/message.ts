import { gql } from 'apollo-server-express'

export const messageTypeDefs = gql`
    type Message {
        id: ID!
        message: String
        timestamp: String
        from: User
        to: User
    }
`
