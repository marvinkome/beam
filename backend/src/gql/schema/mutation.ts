import { gql } from 'apollo-server-express'

export const mutationTypeDefs = gql`
    # INPUTS
    input YoutubeInput {
        name: String!
        id: String!
        image: String
    }

    input RedditInput {
        name: String!
        id: String!
        image: String
    }

    input ArtistsInput {
        name: String!
        id: String!
    }

    input ConnectAccountInput {
        account: ConnectedAccountType!

        # if it's youtube we need subs
        subs: [YoutubeInput]

        # if it's reddit we need subreddits
        subreddits: [RedditInput]

        # if it's spotify we need artists and genres
        artists: [ArtistsInput]
        genres: [String]
    }

    input LocationInput {
        lat: Float
        long: Float
    }

    # MUTATIONS RESPONSES
    type LoginMutationResponse {
        code: String!
        success: Boolean!
        message: String
        token: String
        user: User
    }

    type SendMessageMutationResponse {
        code: String!
        success: Boolean!
        message: String
        sentMessage: Message
    }

    type AddFriendMutationResponse {
        code: String!
        success: Boolean!
        message: String
    }

    type GroupMutationResponse {
        success: Boolean!
        message: String!
        group: Group
    }

    type Mutation {
        # AUTH
        googleLogin(
            token: String!
            inviteToken: String
            youtubeData: [YoutubeInput]
        ): LoginMutationResponse
        facebookLogin(token: String!, inviteToken: String): LoginMutationResponse

        # USER
        connectAccount(input: ConnectAccountInput): Boolean
        disconnectAccount(account: String!): Boolean
        setLocation(location: LocationInput): User
        createInviteLink: String
        addFriend(inviteToken: String!): AddFriendMutationResponse
        addFriendById(friendId: ID!): Boolean
        sendFriendRequest(matchId: ID!): Boolean
        respondToFriendRequest(matchId: ID!, accepted: Boolean!): Boolean

        # GROUP
        createGroup(interestId: ID!): GroupMutationResponse
        joinGroup(groupId: ID!): GroupMutationResponse
        leaveGroup(groupId: ID!): Boolean

        # MESSAGING
        sendMessage(to: ID!, message: String!): SendMessageMutationResponse
        sendMessageToGroup(to: ID!, message: String!): SendMessageMutationResponse
    }
`
