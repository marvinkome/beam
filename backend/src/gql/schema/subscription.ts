import { gql } from 'apollo-server-express'

export const subscriptionTypeDef = gql`
    type Subscription {
        messageSent(friendId: ID, groupId: ID): Message
        lastSeen(friendId: ID!): String
    }
`
