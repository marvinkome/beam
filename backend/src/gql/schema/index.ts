import { queryTypeDef } from './query'
import { mutationTypeDefs } from './mutation'
import { userTypeDefs } from './user'
import { messageTypeDefs } from './message'
import { groupTypeDefs } from './group'
import { subscriptionTypeDef } from './subscription'

export const typeDefs = [
    queryTypeDef,
    mutationTypeDefs,
    userTypeDefs,
    groupTypeDefs,
    messageTypeDefs,
    subscriptionTypeDef,
]
