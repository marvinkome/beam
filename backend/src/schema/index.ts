import { queryTypeDef } from './query'
import { mutationTypeDefs } from './mutation'
import { userTypeDefs } from './user'
import { messageTypeDefs } from './message'
import { groupTypeDefs } from './group'
import { conversationTypeDef } from './converation'
import { subscriptionTypeDef } from './subscription'

export const typeDefs = [
    queryTypeDef,
    mutationTypeDefs,
    userTypeDefs,
    groupTypeDefs,
    messageTypeDefs,
    conversationTypeDef,
    subscriptionTypeDef,
]
