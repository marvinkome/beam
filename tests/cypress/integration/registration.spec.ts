import schema from "../fixtures/schema.json"
const serverUrl = Cypress.env("serverUrl")

describe.skip("Landing page", () => {
    beforeEach(() => {
        // fixtures
        cy.fixture("connect-accounts/youtube.json").as("youtubeSubsJSON")

        // setup mock server
        cy.server()
        cy.mockGraphql({ schema, endpoint: `${serverUrl}/graphql` }).as("mockGraphqlOps")

        cy.visit("/")
    })

    it.skip("loads page <snapshot>", () => {
        // @ts-ignore
        cy.get(".landing-page").toMatchSnapshot()
    })

    it("sign up with google <e2e>", () => {
        // clean database and visit

        // get button and wait for it to be enabled
        cy.get(".btn.btn-primary", { timeout: 5000 })
            .should("have.length", 1)
            .should("not.have.class", "disabled")

        // mock google login
        cy.window()
            .its("gapi")
            .then((gapi) => {
                // expect object to be defined
                expect(gapi).to.have.property("auth2")

                // create mock response
                const res = {
                    getBasicProfile: () => ({
                        getId: () => "randomGoogleId",
                        getImageUrl: () => "image-url",
                        getEmail: () => "johndoe@gmail.com",
                        getName: () => "John Doe",
                        getGivenName: () => "John",
                        getFamilyName: () => "Doe",
                    }),
                    getAuthResponse: () => ({
                        id_token: "id_token",
                        access_token: "my-access-token",
                    }),
                }

                // mock sign in method
                cy.stub(gapi.auth2, "getAuthInstance").returns({
                    signIn: () => new Promise((resolve) => resolve(res)),
                })
            })

        cy.get(".btn.btn-primary").click()

        // expect redirect
        cy.url()
            .should("include", "/app/onboarding")
            .then(() => {
                expect(localStorage.getItem("Beam_Auth_Token_Dev")).exist
            })
    })

    it.skip("sign up with google", () => {
        // mock login mutation
        cy.mockGraphqlOps({
            operations: {
                GoogleLogin: {
                    googleLogin: {
                        success: true,
                        token: "auth-token",
                        user: {
                            id: "user-id",
                            email: "user@test.com",
                            createdAt: "2020-09-22T13:33:59.825Z",
                            hasConnectedAccount: false,
                        },
                    },
                },
            },
        })

        // mock connect youtube mutation
        cy.mockGraphqlOps({
            operations: {
                ConnectAccount: {
                    connectAccount: true,
                },
            },
        })

        // mock youtube subscriptions
        cy.route(
            /https:\/\/www.googleapis.com\/youtube\/v3\/subscriptions*/,
            "@youtubeSubsJSON"
        ).as("youtubeSubsAPI")

        // get button and wait for it to be enabled
        cy.get(".btn.btn-primary", { timeout: 5000 })
            .should("have.length", 1)
            .should("not.have.class", "disabled")

        // mock google login
        cy.window()
            .its("gapi")
            .then((gapi) => {
                // expect object to be defined
                expect(gapi).to.have.property("auth2")

                // create mock response
                const res = {
                    getBasicProfile: () => ({
                        getId: () => "mock-google-id",
                        getImageUrl: () => "image-url",
                        getEmail: () => "email@test.com",
                        getName: () => "name",
                        getGivenName: () => "given name",
                        getFamilyName: () => "family name",
                    }),
                    getAuthResponse: () => ({
                        id_token: "id_token",
                        access_token: "my-access-token",
                    }),
                }

                // mock sign in method
                cy.stub(gapi.auth2, "getAuthInstance").returns({
                    signIn: () => new Promise((resolve) => resolve(res)),
                })
            })

        cy.get(".btn.btn-primary").click()

        // expect redirect
        cy.url().should("include", "/app/onboarding")
    })

    it.skip("login with google", () => {
        // mock login mutation
        cy.mockGraphqlOps({
            operations: {
                GoogleLogin: {
                    googleLogin: {
                        success: true,
                        token: "auth-token",
                        user: {
                            id: "user-id",
                            email: "user@test.com",
                            createdAt: "2020-09-22T13:33:59.825Z",
                            hasConnectedAccount: true,
                        },
                    },
                },
            },
        })

        // get login link and wait for it to be enabled
        cy.get(".header-cta .link-button", { timeout: 5000 })
            .should("have.length", 1)
            .should("not.have.class", "disabled")

        // mock google login
        cy.window()
            .its("gapi")
            .then((gapi) => {
                // expect object to be defined
                expect(gapi).to.have.property("auth2")

                // create mock response
                const res = {
                    getBasicProfile: () => ({
                        getId: () => "mock-google-id",
                        getImageUrl: () => "image-url",
                        getEmail: () => "email@test.com",
                        getName: () => "name",
                        getGivenName: () => "given name",
                        getFamilyName: () => "family name",
                    }),
                    getAuthResponse: () => ({
                        id_token: "id_token",
                        access_token: "my-access-token",
                    }),
                }

                // mock sign in method
                cy.stub(gapi.auth2, "getAuthInstance").returns({
                    signIn: () => new Promise((resolve) => resolve(res)),
                })
            })

        cy.get(".header-cta .link-button").click()

        // expect redirect
        cy.url().should("include", "/app/chats")
    })
})

describe("Registration flow", () => {
    beforeEach(() => {
        // fixtures
        cy.fixture("connect-accounts/youtube.json").as("youtubeSubsJSON")

        cy.setup("/", {
            onBeforeLoad(win) {
                // stub browser permissions
                cy.stub(win.navigator.geolocation, "getCurrentPosition", (cb) => {
                    return cb({
                        coords: {
                            accuracy: 51528,
                            altitude: null,
                            altitudeAccuracy: null,
                            heading: null,
                            latitude: 6.4487423999999995,
                            longitude: 3.4701312,
                            speed: null,
                        },
                        timestamp: 1600955261472,
                    })
                })

                // @ts-ignore
                cy.stub(win.Notification, "requestPermission").resolves("granted")
            },
        })
    })

    it("Complete sign up and onboarding flow without match", () => {
        // ========== SETUP MOCKS ============= //
        // mock youtube subscriptions
        cy.route(
            /https:\/\/www.googleapis.com\/youtube\/v3\/subscriptions*/,
            "@youtubeSubsJSON"
        ).as("youtubeSubsAPI")

        // get button and wait for it to be enabled
        cy.get(".btn.btn-primary", { timeout: 5000 })
            .should("have.length", 1)
            .should("not.have.class", "disabled")
            // @ts-ignore
            .then(() => cy.get(".landing-page").toMatchSnapshot())

        // mock google login
        cy.window()
            .its("gapi")
            .then((gapi) => {
                // expect object to be defined
                expect(gapi).to.have.property("auth2")

                // create mock response
                const res = {
                    getBasicProfile: () => ({
                        getId: () => "randomGoogleId",
                        getImageUrl: () => "image-url",
                        getEmail: () => "johndoe@gmail.com",
                        getName: () => "John Doe",
                        getGivenName: () => "John",
                        getFamilyName: () => "Doe",
                    }),
                    getAuthResponse: () => ({
                        id_token: "id_token",
                        access_token: "my-access-token",
                    }),
                }

                // mock sign in method
                cy.stub(gapi.auth2, "getAuthInstance").returns({
                    signIn: () => new Promise((resolve) => resolve(res)),
                })
            })

        cy.get(".btn.btn-primary").click()

        // expect redirect
        cy.url()
            .should("include", "/app/onboarding")
            .then(() => {
                expect(localStorage.getItem("Beam_Auth_Token_Dev")).exist
            })

        cy.get("a[href='/app/profile'].btn", { timeout: 5000 }).should("have.length", 1)
    })
})
