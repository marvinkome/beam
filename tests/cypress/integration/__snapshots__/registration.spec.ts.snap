exports[`loads page <snapshot> #0`] = `
<div class="landing-page">
  <div class="header-container">
    <nav class="public-navbar">
      <div>
        <img alt="beam" src="/static/media/beam-logo-dark.f3a266f1.png" />
      </div>
    </nav>
    <div class="header-grid">
      <div class="text-grid">
        <h1>Chat with friends in your city who share your interests</h1>
        <div class="check-list">
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>
              Find people with similar interests on YouTube, Spotify, and Reddit
            </p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>Join a local group for your YouTube subscriptions</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No Ads</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No blue theme</p>
          </div>
        </div>
        <div class="header-cta">
          <button class="btn btn-primary disabled">Sign up with Google</button>
          <p>
            Already have an account?
            <button class="link-button disabled">Log In</button>
          </p>
        </div>
      </div>
      <div class="image-grid">
        <img
          alt="Mobile screenshot of beam"
          src="/static/media/screenshot.67cf2ba9.png"
        />
      </div>
    </div>
  </div>
  <footer class="public-footer">
    <p class="feedback-text">
      If you’ve got any questions, feedback or improvement ideas, we’ve created
      a subreddit (<a href="https://www.reddit.com/r/beam_me/">r/beam_me</a>),
      where you can tell us about it.
    </p>
    <div class="grid-container-footer">
      <div class="grid-1-footer">
        <p><a href="https://usebeam.chat/privacy-policy">Privacy Policy</a></p>
      </div>
      <div class="grid-2-footer">
        <p>
          <a href="https://usebeam.chat/terms-and-condition">Terms of Use</a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Registration flow > Complete sign up and onboarding flow #0`] = `
<div class="landing-page">
  <div class="header-container">
    <nav class="public-navbar">
      <div>
        <img alt="beam" src="/static/media/beam-logo-dark.f3a266f1.png" />
      </div>
    </nav>
    <div class="header-grid">
      <div class="text-grid">
        <h1>Chat with friends in your city who share your interests</h1>
        <div class="check-list">
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>
              Find people with similar interests on YouTube, Spotify, and Reddit
            </p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>Join a local group for your YouTube subscriptions</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No Ads</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No blue theme</p>
          </div>
        </div>
        <div class="header-cta">
          <button class="btn btn-primary disabled">Sign up with Google</button>
          <p>
            Already have an account?
            <button class="link-button disabled">Log In</button>
          </p>
        </div>
      </div>
      <div class="image-grid">
        <img
          alt="Mobile screenshot of beam"
          src="/static/media/screenshot.67cf2ba9.png"
        />
      </div>
    </div>
  </div>
  <footer class="public-footer">
    <p class="feedback-text">
      If you’ve got any questions, feedback or improvement ideas, we’ve created
      a subreddit (<a href="https://www.reddit.com/r/beam_me/">r/beam_me</a>),
      where you can tell us about it.
    </p>
    <div class="grid-container-footer">
      <div class="grid-1-footer">
        <p><a href="https://usebeam.chat/privacy-policy">Privacy Policy</a></p>
      </div>
      <div class="grid-2-footer">
        <p>
          <a href="https://usebeam.chat/terms-and-condition">Terms of Use</a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Registration flow - without match > Complete sign up and onboarding flow #0`] = `
<div class="landing-page">
  <div class="header-container">
    <nav class="public-navbar">
      <div>
        <img alt="beam" src="/static/media/beam-logo-dark.f3a266f1.png" />
      </div>
    </nav>
    <div class="header-grid">
      <div class="text-grid">
        <h1>Chat with friends in your city who share your interests</h1>
        <div class="check-list">
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>
              Find people with similar interests on YouTube, Spotify, and Reddit
            </p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>Join a local group for your YouTube subscriptions</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No Ads</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No blue theme</p>
          </div>
        </div>
        <div class="header-cta">
          <button class="btn btn-primary disabled">Sign up with Google</button>
          <p>
            Already have an account?
            <button class="link-button disabled">Log In</button>
          </p>
        </div>
      </div>
      <div class="image-grid">
        <img
          alt="Mobile screenshot of beam"
          src="/static/media/screenshot.67cf2ba9.png"
        />
      </div>
    </div>
  </div>
  <footer class="public-footer">
    <p class="feedback-text">
      If you’ve got any questions, feedback or improvement ideas, we’ve created
      a subreddit (<a href="https://www.reddit.com/r/beam_me/">r/beam_me</a>),
      where you can tell us about it.
    </p>
    <div class="grid-container-footer">
      <div class="grid-1-footer">
        <p><a href="https://usebeam.chat/privacy-policy">Privacy Policy</a></p>
      </div>
      <div class="grid-2-footer">
        <p>
          <a href="https://usebeam.chat/terms-and-condition">Terms of Use</a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Registration flow > Complete sign up and onboarding flow without match #0`] = `
<div class="landing-page">
  <div class="header-container">
    <nav class="public-navbar">
      <div>
        <img alt="beam" src="/static/media/beam-logo-dark.f3a266f1.png" />
      </div>
    </nav>
    <div class="header-grid">
      <div class="text-grid">
        <h1>Chat with friends in your city who share your interests</h1>
        <div class="check-list">
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>
              Find people with similar interests on YouTube, Spotify, and Reddit
            </p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>Join a local group for your YouTube subscriptions</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No Ads</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No blue theme</p>
          </div>
        </div>
        <div class="header-cta">
          <button class="btn btn-primary">Sign up with Google</button>
          <p>
            Already have an account? <button class="link-button">Log In</button>
          </p>
        </div>
      </div>
      <div class="image-grid">
        <img
          alt="Mobile screenshot of beam"
          src="/static/media/screenshot.67cf2ba9.png"
        />
      </div>
    </div>
  </div>
  <footer class="public-footer">
    <p class="feedback-text">
      If you’ve got any questions, feedback or improvement ideas, we’ve created
      a subreddit (<a href="https://www.reddit.com/r/beam_me/">r/beam_me</a>),
      where you can tell us about it.
    </p>
    <div class="grid-container-footer">
      <div class="grid-1-footer">
        <p><a href="https://usebeam.chat/privacy-policy">Privacy Policy</a></p>
      </div>
      <div class="grid-2-footer">
        <p>
          <a href="https://usebeam.chat/terms-and-condition">Terms of Use</a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Registration flow > Complete sign up and onboarding flow without match #1`] = `
<div class="landing-page">
  <div class="header-container">
    <nav class="public-navbar">
      <div>
        <img alt="beam" src="/static/media/beam-logo-dark.f3a266f1.png" />
      </div>
    </nav>
    <div class="header-grid">
      <div class="text-grid">
        <h1>Chat with friends in your city who share your interests</h1>
        <div class="check-list">
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>
              Find people with similar interests on YouTube, Spotify, and Reddit
            </p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>Join a local group for your YouTube subscriptions</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No Ads</p>
          </div>
          <div class="check">
            <svg
              stroke="currentColor"
              fill="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              class="icon"
              height="1em"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              ></path>
            </svg>
            <p>No blue theme</p>
          </div>
        </div>
        <div class="header-cta">
          <button class="btn btn-primary">Sign up with Google</button>
          <p>
            Already have an account? <button class="link-button">Log In</button>
          </p>
        </div>
      </div>
      <div class="image-grid">
        <img
          alt="Mobile screenshot of beam"
          src="/static/media/screenshot.67cf2ba9.png"
        />
      </div>
    </div>
  </div>
  <footer class="public-footer">
    <p class="feedback-text">
      If you’ve got any questions, feedback or improvement ideas, we’ve created
      a subreddit (<a href="https://www.reddit.com/r/beam_me/">r/beam_me</a>),
      where you can tell us about it.
    </p>
    <div class="grid-container-footer">
      <div class="grid-1-footer">
        <p><a href="https://usebeam.chat/privacy-policy">Privacy Policy</a></p>
      </div>
      <div class="grid-2-footer">
        <p>
          <a href="https://usebeam.chat/terms-and-condition">Terms of Use</a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
